woodwork/table_accessor.py

```diff
@@ -168,6 +170,33 @@ def __getitem__(self, key):
         series.ww.init(**column, use_standard_tags=self.use_standard_tags)
         return series
 
+    def __setitem__(self, col_name, column):
+        series = tuple(pkg.Series for pkg in (pd, dd, ks) if pkg)
+        if not isinstance(column, series):
+            raise ValueError('New column must be of Series type')
+
+        # Don't allow reassigning of index or time index with setitem
+        if self.index == col_name:
+            raise KeyError('Cannot reassign index. Change column name and then use df.ww.set_index to reassign index.')
+        if self.time_index == col_name:
+            raise KeyError('Cannot reassign time index. Change column name and then use df.ww.set_time_index to reassign time index.')
+
+        if column.name is not None and column.name != col_name:
+            warnings.warn(ColumnNameMismatchWarning().get_warning_message(column.name, col_name),
+                          ColumnNameMismatchWarning)
+
+        if column.ww._schema is None:
+            column = init_series(column, use_standard_tags=self.use_standard_tags)
+        elif self.use_standard_tags and not column.ww.use_standard_tags:
+            column.ww._schema['semantic_tags'] |= column.ww.logical_type.standard_tags
+            warnings.warn(f'Standard tags are used by the data frame and have been added to "{col_name}"')
+        elif not self.use_standard_tags and column.ww.use_standard_tags:
+            column.ww._schema['semantic_tags'] -= column.ww.logical_type.standard_tags
+            warnings.warn(f'Standard tags are not used by the data frame and have been removed from "{col_name}"')
```

Comment:
We have a `StandardTagsRemovalWarning` in exceptions.py that should be useful here. Probably worth adding a `StandardTagsAddedWarning`. 

Though it might be worth standardizing it to a single use `StandardTagsChangedWarning` if we can find a good way to update the warning language to work for both.

Reply:
✅ ca1b061

Reply:
Can we also remove the usage of `StandardTagsRemovalWarning` in `schema.py` and use the new warning instead?

Reply:
Hmm, can't seem to find the usage in `schema.py`. Can you attach a link to the lines of code?

Reply:
I meant `schema_column.py`! sorry!

Reply:
https://github.com/alteryx/woodwork/pull/497/files#diff-2f8d2164896536382b984429f46712c1754ab4d361f34672ac3aa81443a2f5f6R123

Reply:
Oh okay, will update

Reply:
✅ 60a98ba